src/alloc/globalalloc.c

```diff
@@ -5,33 +5,145 @@ static struct allocator global_allocator;
 
 #define POOL_SIZE 16 * 1024 * 1024
 
+/** Pool node for metadata */
+struct Node
```

Comment:
You know me. I love documentation. In order to make it easier to maintain this file in the future, I'd like you to document the fields in this struct. It uses the same syntax as the rest of the documentation in C (`/** */`).